### PR TITLE
Introduce macOS Support (Take 2)

### PR DESCRIPTION
**Pull request by capnkenny**
Same as #417

### CODE REVIEW

**Comment by capnkenny** [comment]
To note - until we are ready, this _will not_ cover how to build as well as having CI for macOS. It will be covered in a new PR once I've fixed the Conan package issues

File: src/NovelRT/CMakeLists.txt [hunk 1]
```diff
@@ -169,3 +171,10 @@ if(WIN32)
   file(GLOB libs RELATIVE ${CMAKE_BINARY_DIR}/thirdparty ${CMAKE_BINARY_DIR}/thirdparty/*.dll)
   set_target_properties(Engine PROPERTIES RESOURCE "${libs}")
 endif()
+if(APPLE)
+  add_custom_command(TARGET Engine POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory
+          ${CMAKE_BINARY_DIR}/thirdparty
+          $<TARGET_FILE_DIR:Engine>)
+  file(GLOB libs RELATIVE ${CMAKE_BINARY_DIR}/thirdparty ${CMAKE_BINARY_DIR}/thirdparty/*.dylib)
+  set_target_properties(Engine PROPERTIES RESOURCE "${libs}")
+endif()
```
Comment (RubyNova) [inline src/NovelRT/CMakeLists.txt:180]: Missing EOF

File: src/NovelRT/CMakeLists.txt [hunk 2]
```diff
@@ -169,3 +171,10 @@ if(WIN32)
   file(GLOB libs RELATIVE ${CMAKE_BINARY_DIR}/thirdparty ${CMAKE_BINARY_DIR}/thirdparty/*.dll)
   set_target_properties(Engine PROPERTIES RESOURCE "${libs}")
 endif()
+if(APPLE)
+  add_custom_command(TARGET Engine POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory
+          ${CMAKE_BINARY_DIR}/thirdparty
+          $<TARGET_FILE_DIR:Engine>)
```
Comment (RubyNova) [inline src/NovelRT/CMakeLists.txt:177]: Can't this use copy_build_products instead?

**Review by RubyNova — CHANGES_REQUESTED** [review]
One question and one change from me.

File: src/NovelRT/CMakeLists.txt (hunk 2 above)
Comment (capnkenny) [reply src/NovelRT/CMakeLists.txt:177]: Not that I'm aware of - if so, I'd propose a separate PR for it considering it's an existing issue at that point, not something introduced by macOS support :)

File: src/NovelRT/CMakeLists.txt (hunk 1 above)
Comment (capnkenny) [reply src/NovelRT/CMakeLists.txt]: Should be fixed now

File: src/NovelRT/CMakeLists.txt (hunk 2 above)
Comment (capnkenny) [reply src/NovelRT/CMakeLists.txt:177]: Just adding another thought - we cannot do this as we don't have a proper target to attach the DLLs/dylibs to (they get attached to Engine after being initially copied so its accessible to anything dependent on Engine)

**Review by RubyNova — APPROVED** [review]
LGTM

**Pull request merged**